File: frontend/src/app/shared/model/jogador.ts
export class Jogador{
  public id?: string;
  public nome: string;
  public especial: boolean;
  public score: number

  constructor(id?: string, jogador: Partial<Jogador> = {}){
    this.id = id
    this.nome = jogador.nome || '';
    this.especial = jogador.especial || false;
    this.score = jogador.score || 0;
  }

  toString(): string{
    return this.nome
  }
}

File: backend/src/main/java/com/matchmanager/model/Jogador.java
package com.matchmanager.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
public class Jogador {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private boolean especial;
    private Integer score;
    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name = "match_id")
    @JsonIgnore
    private Match match;

    public Long getId() {
        return id;
    }


    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public boolean isEspecial() {
        return especial;
    }

    public void setEspecial(boolean Especial) {
        this.especial = Especial;
    }

    public Integer getScore() {
        return score;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public Match getMatch() {
        return match;
    }

    public void setMatch(Match match) {
        this.match = match;
    }
}
